build-system/compile/compile.js

```diff
@@ -236,15 +236,27 @@ function compile(
     }
     externs.push('build-system/externs/amp.multipass.extern.js');
 
+    // Normally setting this server-side experiment flag would be handled by
+    // the release process automatically. Since this experiment is actually on the
+    // build system instead of runtime, we never run it through babel and therefore
+    // must compute it here.
+    const STRICT_COMPILATION =
+      argv.define_experiment_constant &&
+      argv.define_experiment_constant === 'STRICT_COMPILATION';
+    let target = STRICT_COMPILATION ? 'ECMASCRIPT5_STRICT' : 'ECMASCRIPT5';
+    if (argv.esm) {
+      // Do not transpile down to ES5 if running with `--esm`, since we do
+      // limited transpilation in Babel.
+      target = 'NO_TRANSPILE';
+    }
+
     /* eslint "google-camelcase/google-camelcase": 0*/
     const compilerOptions = {
       compilation_level: options.compilationLevel || 'SIMPLE_OPTIMIZATIONS',
       // Turns on more optimizations.
       assume_function_wrapper: true,
       language_in: 'ECMASCRIPT_2020',
-      // Do not transpile down to ES5 if running with `--esm`, since we do
-      // limited transpilation in Babel.
-      language_out: argv.esm ? 'NO_TRANSPILE' : 'ECMASCRIPT5',
+      language_out: target,
```

Comment:
Based on the above suggestion:

```suggestion
      language_out: language,
```